# EDA Dataset: NYC Restaurant Food Order & Delivery
## Análisis de criterios de selección

Este notebook recoge el análisis exploratorio del dataset NYC Restaurant 
Food Order & Delivery como parte del proceso de selección del dataset 
para el TFG de predicción de demanda en pizzerías con delivery.

In [1]:
import pandas as pd

# Carga del dataset
df = pd.read_csv('../../data/0_seleccion_dataset/kaggle_NYC_restaurants_data/food_order.csv')

print(f"Dataset cargado correctamente")

print("--- CRITERIO 1: VOLUMEN ")
print(f"Registros totales: {df.shape[0]}")
print(f"\nRestaurantes distintos: {df['restaurant_name'].nunique()}")
print(f"Tipos de cocina distintos: {df['cuisine_type'].nunique()}")

Dataset cargado correctamente
--- CRITERIO 1: VOLUMEN 
Registros totales: 1898

Restaurantes distintos: 178
Tipos de cocina distintos: 14


## Criterio 1: Volumen
El dataset solo contiene 1.898 registros siendo insuficiente
para entrenar el modelo.

In [2]:
print("--- CRITERIO 2: GRANULARIDAD TEMPORAL ")
print(f"Columnas disponibles: {df.columns.tolist()}")
print(f"¿Tiene fecha? {'date' in df.columns}")
print(f"¿Tiene hora? {'time' in df.columns or 'hour' in df.columns}")
print(f"\nVariable temporal disponible: day_of_the_week")
print(f"Valores únicos: {df['day_of_the_week'].unique()}")

--- CRITERIO 2: GRANULARIDAD TEMPORAL 
Columnas disponibles: ['order_id', 'customer_id', 'restaurant_name', 'cuisine_type', 'cost_of_the_order', 'day_of_the_week', 'rating', 'food_preparation_time', 'delivery_time']
¿Tiene fecha? False
¿Tiene hora? False

Variable temporal disponible: day_of_the_week
Valores únicos: <ArrowStringArray>
['Weekend', 'Weekday']
Length: 2, dtype: str


## Criterio 2: Granularidad temporal
El dataset no dispone de fecha ni hora de los pedidos. La única 
información temporal es si el pedido se realizó en fin de semana 
o entre semana, con solo dos valores posibles: Weekend y Weekday. 
Esto hace imposible construir una serie temporal.

In [3]:
print("--- CRITERIO 3: RANGO TEMPORAL ")
print(f"¿Tiene columna de fecha? {'date' in df.columns}")
print(f"\nDistribución por tipo de día:")
print(df['day_of_the_week'].value_counts())

--- CRITERIO 3: RANGO TEMPORAL 
¿Tiene columna de fecha? False

Distribución por tipo de día:
day_of_the_week
Weekend    1351
Weekday     547
Name: count, dtype: int64


## Criterio 3: Rango temporal
Al no disponer de fecha real no es posible determinar el rango 
temporal del dataset ni construir una serie temporal ordenada 
en el tiempo.

In [4]:
print("--- CRITERIO 4: CALIDAD DE DATOS ")
print(f"Valores nulos:\n{df.isnull().sum()}")
print(f"\nDuplicados: {df.duplicated().sum()}")

--- CRITERIO 4: CALIDAD DE DATOS 
Valores nulos:
order_id                 0
customer_id              0
restaurant_name          0
cuisine_type             0
cost_of_the_order        0
day_of_the_week          0
rating                   0
food_preparation_time    0
delivery_time            0
dtype: int64

Duplicados: 0


## Criterio 4: Calidad de los datos
El dataset no presenta valores nulos ni registros duplicados.

In [5]:
print("--- CRITERIO 5: ESPECIFICIDAD DE RESTAURACIÓN ")
print(f"Restaurantes distintos: {df['restaurant_name'].nunique()}")
print(f"Tipos de cocina: {df['cuisine_type'].nunique()}")
print(f"\nTipos de cocina disponibles:")
print(df['cuisine_type'].value_counts())

--- CRITERIO 5: ESPECIFICIDAD DE RESTAURACIÓN 
Restaurantes distintos: 178
Tipos de cocina: 14

Tipos de cocina disponibles:
cuisine_type
American          584
Japanese          470
Italian           298
Chinese           215
Mexican            77
Indian             73
Middle Eastern     49
Mediterranean      46
Thai               19
French             18
Southern           17
Korean             13
Spanish            12
Vietnamese          7
Name: count, dtype: int64


## Criterio 5: Especificidad de restauración
El dataset incluye 178 restaurantes distintos de 14 tipos de 
cocina diferentes en Nueva York. No representa un único 
establecimiento ni está orientado específicamente a pizzerías 
con delivery.

## Verificación de criterios de evaluación

- ❌ **Volumen:** 1.898 registros repartidos entre 178 restaurantes 
  y 14 tipos de cocina distintos.
- ❌ **Fecha y hora:** sin fecha ni hora, solo indica 
  si el pedido es de fin de semana o entre semana.
- ❌ **Rango temporal:** imposible determinarlo al no disponer 
  de fecha real.
- ✅ **Calidad datos:** sin valores nulos ni duplicados.
- ❌ **Especificidad:** 178 restaurantes de 14 tipos de cocina 
  distintos, no representa un único establecimiento ni es 
  específico de pizzerías.

## Conclusión: dataset descartado

El análisis confirma que este dataset no cumple los criterios 
mínimos de selección.

La ausencia de fecha y hora es el criterio determinante para 
el descarte, ya que hace imposible construir una serie temporal 
y por tanto entrenar cualquier modelo predictivo.

**Decisión: dataset descartado.**